lib/nats.js

```diff
@@ -1166,9 +1166,16 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         opt_callback = subject;
         subject = undefined;
     }
-    if (!msg) {
-        msg = EMPTY;
+
+    if (!this.options.json) {
+        if (!msg) {
+            msg = EMPTY;
+        }
+    } else if (msg === undefined) {
```

Comment:
I think the correct thing to do here is to make msg `null` - On other clients where the value is required, requests/publishes will have to provide null/nil, which will serialize as `null` in the JSON. We need to treat the default message to be `null` in the case of json.

The publish and request need to add a test where they elide the payload, and the subscribed value is expected to have been a serialized `null` json.